Framework/DataHandling/src/LoadNexusLogs.cpp

```diff
@@ -654,5 +656,33 @@ LoadNexusLogs::createTimeSeries(::NeXus::File &file,
   }
 }
 
+/**
+  * Checks whether the specified character is invalid or a control
+  * character. If it is invalid (i.e. negative) or a control character
+  * the method returns true. If it is valid and not a control character
+  * it returns false. Additionally if the character is invalid is
+  * logs a warning with the property name so users are aware.
+  *
+  * @param c :: Character to check
+  * @param propName :: The name of the property currently being checked for
+  *logging
+  * @param log :: Reference to logger to print out to
+  * @return :: True if control character OR invalid. Else False
+  */
+bool LoadNexusLogs::isControlValue(const char &c, const std::string &propName,
+                                   Kernel::Logger &log) {
+  // Have to check it falls within range accepted by c style check
+  if (c <= -1) {
```

Comment:
Your right - looking around whilst for `int` and other larger types the default sign is specified in the standard for a `char` it is implementation defined (based on architecture)

In this case if we cast up from an unsigned `char` where that is used on that architecture to a `signed int` we should be able to both hold the value and sign regardless. Additionally if we do this we need to check the value is not greater than 255 as well (if using unsigned types)

Looking again at the compiler output this doesn't add any extra overhead but should ensure we are doing the correct thing : https://godbolt.org/g/Y5ewMz